### ISSUE TITLE
Pymunk fails with default cffi

### ISSUE THREAD

**Issue by manuelli** [viblo/pymunk#171]
(edited). See the comment below for a more minimal example.

**Comment by manuelli** [comment]
Actually I have investigated a bit more and the issue doesn't seem related to the nvidia driver. To test this I created a Python 2.7.12 virtualenv on Ubuntu 16.04. The only thing that was installed is `pymunk`, i.e.

```
(key_dynam_2) manuelli@paladin-23:~/code/key_dynam$ pip freeze
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
cffi==1.13.1
pycparser==2.19
pymunk==5.5.0
```
But running the unittest yields the same result as above. Is there some system installed library that could be causing a problem, even inside a virtualenv? Outcome of `python -m pymunk.tests test`

```
(key_dynam_2) manuelli@paladin-23:~/code/key_dynam$ python -m pymunk.tests test
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBNewForExtents' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBNewForCircle' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBIntersects' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBContainsBB' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBContainsVect' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBMerge' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBExpand' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBCenter' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBArea' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBMergedArea' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBSegmentQuery' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBIntersectsSegment' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/cffi/cparser.py:164: UserWarning: Declaration of global variable '_cpBBClampVect' in cdef() should be marked 'extern' for consistency (or possibly 'static' in API mode)
  "'static' in API mode)" % (decl.name,))
Loading chipmunk for Linux (64bit) [/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/pymunk/libchipmunk.so]
('__init__', 'pymunk.tests', '/home/manuelli/virtualenvs/key_dynam_2/local/lib/python2.7/site-packages/pymunk/tests/__init__.pyc')
Skipping _chipmunk_cffi
Skipping _chipmunk_cffi_abi
Skipping libchipmunk
Skipping matplotlib_util
Skipping pygame_util
Skipping pyglet_util
Skipping tests
Skipping transform
testContactPointSet (test_arbiter.UnitTestArbiter) ... ok
testFriction (test_arbiter.UnitTestArbiter) ... ok
testImpulse (test_arbiter.UnitTestArbiter) ... ok
testIsFirstContact (test_arbiter.UnitTestArbiter) ... ok
testIsRemoval (test_arbiter.UnitTestArbiter) ... ok
testRestitution (test_arbiter.UnitTestArbiter) ... ok
testShapes (test_arbiter.UnitTestArbiter) ... ok
testSurfaceVelocity (test_arbiter.UnitTestArbiter) ... ok
testTotalKE (test_arbiter.UnitTestArbiter) ... ok
testCircleBB (test_shape.UnitTestCircle) ... ok
testCircleNoBody (test_shape.UnitTestCircle) ... ok
testOffset (test_shape.UnitTestCircle) ... ok
testOffsetUnsafe (test_shape.UnitTestCircle) ... ok
testPickle (test_shape.UnitTestCircle) ... ok
testRadius (test_shape.UnitTestCircle) ... ok
testRadiusUnsafe (test_shape.UnitTestCircle) ... ok
testBB (test_shape.UnitTestPoly) ... ok
testCreateBox (test_shape.UnitTestPoly) ... ok
testInit (test_shape.UnitTestPoly) ... ok
testPickle (test_shape.UnitTestPoly) ... ok
testRadius (test_shape.UnitTestPoly) ... ok
testRadiusUnsafe (test_shape.UnitTestPoly) ... ok
testVertices (test_shape.UnitTestPoly) ... ok
testVerticesUnsafe (test_shape.UnitTestPoly) ... ok
testBB (test_shape.UnitTestSegment) ... ok
testPickle (test_shape.UnitTestSegment) ... ok
testProperties (test_shape.UnitTestSegment) ... ok
testPropertiesUnsafe (test_shape.UnitTestSegment) ... ok
testSegmentSegmentCollision (test_shape.UnitTestSegment) ... ok
testSetNeighbors (test_shape.UnitTestSegment) ... ok
testArea (test_shape.UnitTestShape) ... ok
testCenterOfGravity (test_shape.UnitTestShape) ... ok
testCollisionType (test_shape.UnitTestShape) ... ok
testDensity (test_shape.UnitTestShape) ... ok
testElasticity (test_shape.UnitTestShape) ... ok
testFilter (test_shape.UnitTestShape) ... ok
testFriction (test_shape.UnitTestShape) ... ok
testMass (test_shape.UnitTestShape) ... ok
testMoment (test_shape.UnitTestShape) ... ok
testNoBody (test_shape.UnitTestShape) ... ok
testPickle (test_shape.UnitTestShape) ... ok
testPointQuery (test_shape.UnitTestShape) ... ok
testRemoveBody (test_shape.UnitTestShape) ... ok
testSegmentQuery (test_shape.UnitTestShape) ... ok
testSensor (test_shape.UnitTestShape) ... ok
testShapesCollide (test_shape.UnitTestShape) ... ok
testSpace (test_shape.UnitTestShape) ... ok
testSurfaceVelocity (test_shape.UnitTestShape) ... ok
testSwitchBody (test_shape.UnitTestShape) ... ok
testAnglesDegrees (test_vec2d.UnitTestVec2d) ... ok
testAnglesRadians (test_vec2d.UnitTestVec2d) ... ok
testComparison (test_vec2d.UnitTestVec2d) ... ok
testCreationAndAccess (test_vec2d.UnitTestVec2d) ... ok
testCross (test_vec2d.UnitTestVec2d) ... ok
testHighLevel (test_vec2d.UnitTestVec2d) ... ok
testInplace (test_vec2d.UnitTestVec2d) ... ok
testLength (test_vec2d.UnitTestVec2d) ... ok
testMath (test_vec2d.UnitTestVec2d) ... ok
testPickle (test_vec2d.UnitTestVec2d) ... ok
testReverseMath (test_vec2d.UnitTestVec2d) ... ok
testUnary (test_vec2d.UnitTestVec2d) ... ok
testCreation (test_bb.UnitTestBB) ... ok
testMerge (test_bb.UnitTestBB) ... ok
testMethods (test_bb.UnitTestBB) ... ok
testPickle (test_bb.UnitTestBB) ... ok
testManyBoxCrash (test_common.UnitTestBugs) ... ok
testNoStaticShape (test_common.UnitTestBugs) ... ok
testAreaHelpers (test_common.UnitTestGeneral) ... ok
testGeneral (test_common.UnitTestGeneral) ... ok
testMomentHelpers (test_common.UnitTestGeneral) ... ok
testAddRemove (test_space.UnitTestSpace) ... ok
testAddRemoveInStep (test_space.UnitTestSpace) ... ok
testBBQuery (test_space.UnitTestSpace) ... ok
testBBQuerySensor (test_space.UnitTestSpace) ... ok
testCollisionHandlerBegin (test_space.UnitTestSpace) ... ok
testCollisionHandlerBeginNoReturn (test_space.UnitTestSpace) ... ok
testCollisionHandlerPostSolve (test_space.UnitTestSpace) ... ok
testCollisionHandlerPreSolve (test_space.UnitTestSpace) ... ok
testCollisionHandlerPreSolveNoReturn (test_space.UnitTestSpace) ... ok
testCollisionHandlerSeparate (test_space.UnitTestSpace) ... ok
testCopyMethods (test_space.UnitTestSpace) ... ok
testDebugDraw (test_space.UnitTestSpace) ... FAIL
testDefaultCollisionHandler (test_space.UnitTestSpace) ... ok
testPointQuery (test_space.UnitTestSpace) ... ok
testPointQueryNearest (test_space.UnitTestSpace) ... ok
testPointQueryNearestSensor (test_space.UnitTestSpace) ... ok
testPointQueryNearestWithShapeFilter (test_space.UnitTestSpace) ... ok
testPointQuerySensor (test_space.UnitTestSpace) ... ok
testPostStepCallback (test_space.UnitTestSpace) ... ok
testProperties (test_space.UnitTestSpace) ... ok
testReindexShape (test_space.UnitTestSpace) ... ok
testReindexShapesForBody (test_space.UnitTestSpace) ... ok
testReindexStatic (test_space.UnitTestSpace) ... ok
testReindexStaticCollision (test_space.UnitTestSpace) ... ok
testRemoveInStep (test_space.UnitTestSpace) ... ok
testSegmentQuery (test_space.UnitTestSpace) ... ok
testSegmentQueryFirst (test_space.UnitTestSpace) ... ok
testSegmentQueryFirstSensor (test_space.UnitTestSpace) ... ok
testSegmentQuerySensor (test_space.UnitTestSpace) ... ok
testShapeQuery (test_space.UnitTestSpace) ... ok
testShapeQuerySensor (test_space.UnitTestSpace) ... ok
testStaticPointQueries (test_space.UnitTestSpace) ... ok
testStaticSegmentQueries (test_space.UnitTestSpace) ... ok
testThreaded (test_space.UnitTestSpace) ... ok
testWildcardCollisionHandler (test_space.UnitTestSpace) ... ok
testIdentity (test_transform.UnitTestTransform) ... ok
testInit (test_transform.UnitTestTransform) ... ok
testPickle (test_transform.UnitTestTransform) ... ok
testCoordinateConversion (test_body.UnitTestBody) ... ok
testDynamic (test_body.UnitTestBody) ... ok
testEachArbiters (test_body.UnitTestBody) ... ok
testForce (test_body.UnitTestBody) ... ok
testGetConstraints (test_body.UnitTestBody) ... ok
testGetShapes (test_body.UnitTestBody) ... ok
testImpulse (test_body.UnitTestBody) ... ok
testKinematic (test_body.UnitTestBody) ... ok
testKineticEnergy (test_body.UnitTestBody) ... ok
testMassMomentFromShape (test_body.UnitTestBody) ... ok
testPickle (test_body.UnitTestBody) ... 1
2
3
4
5
6
ok
testPositionFunction (test_body.UnitTestBody) ... ok
testProperties (test_body.UnitTestBody) ... ok
testRepr (test_body.UnitTestBody) ... ok
testSleep (test_body.UnitTestBody) ... ok
testSleepWithGroup (test_body.UnitTestBody) ... ok
testStatic (test_body.UnitTestBody) ... ok
testVelocityConversion (test_body.UnitTestBody) ... ok
testVelocityFunction (test_body.UnitTestBody) ... ok
testConstants (test_shape_filter.UnitTestShapeFilter) ... ok
testEq (test_shape_filter.UnitTestShapeFilter) ... ok
testInit (test_shape_filter.UnitTestShapeFilter) ... ok
testPickle (test_shape_filter.UnitTestShapeFilter) ... ok
test_convex_decomposition (test_autogeometry.UnitTestAutoGeometry) ... ok
test_is_closed (test_autogeometry.UnitTestAutoGeometry) ... ok
test_march_hard (test_autogeometry.UnitTestAutoGeometry) ... FAIL
test_march_soft (test_autogeometry.UnitTestAutoGeometry) ... FAIL
test_simplify_curves (test_autogeometry.UnitTestAutoGeometry) ... ok
test_simplify_vertexes (test_autogeometry.UnitTestAutoGeometry) ... ok
test_to_convex_hull (test_autogeometry.UnitTestAutoGeometry) ... ok
test_collect_segment (test_autogeometry.UnitTestPolylineSet) ... ok
testA (test_constraint.UnitTestConstraint) ... ok
testActivate (test_constraint.UnitTestConstraint) ... ok
testB (test_constraint.UnitTestConstraint) ... ok
testCollideBodies (test_constraint.UnitTestConstraint) ... ok
testErrorBias (test_constraint.UnitTestConstraint) ... ok
testImpulse (test_constraint.UnitTestConstraint) ... ok
testMaxBias (test_constraint.UnitTestConstraint) ... ok
testMaxForce (test_constraint.UnitTestConstraint) ... ok
testPickle (test_constraint.UnitTestConstraint) ... ok
testDamping (test_constraint.UnitTestDampedRotarySpring) ... ok
testPickle (test_constraint.UnitTestDampedRotarySpring) ... ok
testRestAngle (test_constraint.UnitTestDampedRotarySpring) ... ok
testStiffness (test_constraint.UnitTestDampedRotarySpring) ... ok
testAnchor (test_constraint.UnitTestDampedSpring) ... ok
testDamping (test_constraint.UnitTestDampedSpring) ... ok
testPickle (test_constraint.UnitTestDampedSpring) ... ok
testRestLength (test_constraint.UnitTestDampedSpring) ... ok
testStiffness (test_constraint.UnitTestDampedSpring) ... ok
testPhase (test_constraint.UnitTestGearJoint) ... ok
testPickle (test_constraint.UnitTestGearJoint) ... ok
testRatio (test_constraint.UnitTestGearJoint) ... ok
testAnchor (test_constraint.UnitTestGrooveJoint) ... ok
testGroove (test_constraint.UnitTestGrooveJoint) ... ok
testPickle (test_constraint.UnitTestGrooveJoint) ... ok
testAnchor (test_constraint.UnitTestPinJoint) ... ok
testDistane (test_constraint.UnitTestPinJoint) ... ok
testPickle (test_constraint.UnitTestPinJoint) ... ok
testAnchorByAnchor (test_constraint.UnitTestPivotJoint) ... ok
testAnchorByPivot (test_constraint.UnitTestPivotJoint) ... ok
testPickle (test_constraint.UnitTestPivotJoint) ... ok
testAngle (test_constraint.UnitTestRatchetJoint) ... ok
testPhase (test_constraint.UnitTestRatchetJoint) ... ok
testPickle (test_constraint.UnitTestRatchetJoint) ... ok
testRatchet (test_constraint.UnitTestRatchetJoint) ... ok
testMax (test_constraint.UnitTestRotaryLimitJoint) ... ok
testMin (test_constraint.UnitTestRotaryLimitJoint) ... ok
testPickle (test_constraint.UnitTestRotaryLimitJoint) ... ok
testPickle (test_constraint.UnitTestSimleMotor) ... ok
testSimpleMotor (test_constraint.UnitTestSimleMotor) ... ok
testAnchor (test_constraint.UnitTestSlideJoint) ... ok
testMax (test_constraint.UnitTestSlideJoint) ... ok
testMin (test_constraint.UnitTestSlideJoint) ... ok
testPickle (test_constraint.UnitTestSlideJoint) ... ok

======================================================================
FAIL: testDebugDraw (test_space.UnitTestSpace)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/manuelli/virtualenvs/key_dynam_2/lib/python2.7/site-packages/pymunk/tests/test_space.py", line 729, in testDebugDraw
    self.assertEqual(msg, new_out.getvalue())
AssertionError: "('draw_circle', (Vec2d(0.0, 0.0), 0.0, 5.0, SpaceDebugColor(r=44.0, g=62.0, b=80.0, a=255.0), SpaceDebugColor(r=52.0, g=152.0, b=219.0, a=255.0)))\n" != "('draw_circle', (Vec2d(0.0, 0.0), 0.0, 5.0, SpaceDebugColor(r=44.0, g=62.0, b=0.0, a=0.0), SpaceDebugColor(r=52.0, g=152.0, b=0.0, a=0.0)))\n"

======================================================================
FAIL: test_march_hard (test_autogeometry.UnitTestAutoGeometry)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/manuelli/virtualenvs/key_dynam_2/lib/python2.7/site-packages/pymunk/tests/test_autogeometry.py", line 143, in test_march_hard
    self.assertEqual(segments, expected)
AssertionError: Lists differ: [((1.5, 0.0), (1.5, 0.0)), ((3... != [((1.5, 1.0), (1.5, 0.0)), ((3...

First differing element 0:
((1.5, 0.0), (1.5, 0.0))
((1.5, 1.0), (1.5, 0.0))

Second list contains 2 additional elements.
First extra element 12:
((5.0, 4.5), (6.0, 4.5))

Diff is 1294 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_march_soft (test_autogeometry.UnitTestAutoGeometry)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/manuelli/virtualenvs/key_dynam_2/lib/python2.7/site-packages/pymunk/tests/test_autogeometry.py", line 101, in test_march_soft
    self.assertEqual(segments, expected)
AssertionError: Lists differ: [((1.5, 0.0), (1.5, 0.0)), ((3... != [((1.5, 1.0), (1.5, 0.0)), ((3...

First differing element 0:
((1.5, 0.0), (1.5, 0.0))
((1.5, 1.0), (1.5, 0.0))

Second list contains 1 additional elements.
First extra element 12:
((1.5, 6.0), (1.5, 5.0))

Diff is 1211 characters long. Set self.maxDiff to None to see it.

----------------------------------------------------------------------
Ran 182 tests in 0.262s

FAILED (failures=3)

```

**Comment by manuelli** [comment]
It seems that `cffi` is the problem. If I peg it to `cffi==1.12.3` and re-run the tests they work fine. So maybe `pymunk` should peg it's dependency to a compatible version of `cffi`. These tests were performed in a python 2.7.12 virtualenv on Ubuntu 18.04.

```
(pymunk) manuelli@paladin-44:~/software/virtualenv$ pip freeze
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
cffi==1.12.3
pycparser==2.19
pymunk==5.5.0
(pymunk) manuelli@paladin-44:~/software/virtualenv$ python -m pymunk.tests test
Loading chipmunk for Linux (64bit) [/home/manuelli/software/virtualenv/pymunk/local/lib/python2.7/site-packages/pymunk/libchipmunk.so]
('__init__', 'pymunk.tests', '/home/manuelli/software/virtualenv/pymunk/local/lib/python2.7/site-packages/pymunk/tests/__init__.pyc')
Skipping _chipmunk_cffi
Skipping _chipmunk_cffi_abi
Skipping libchipmunk
Skipping matplotlib_util
Skipping pygame_util
Skipping pyglet_util
Skipping tests
Skipping transform
testIdentity (test_transform.UnitTestTransform) ... ok
testInit (test_transform.UnitTestTransform) ... ok
testPickle (test_transform.UnitTestTransform) ... ok
testAddRemove (test_space.UnitTestSpace) ... ok
testAddRemoveInStep (test_space.UnitTestSpace) ... ok
testBBQuery (test_space.UnitTestSpace) ... ok
testBBQuerySensor (test_space.UnitTestSpace) ... ok
testCollisionHandlerBegin (test_space.UnitTestSpace) ... ok
testCollisionHandlerBeginNoReturn (test_space.UnitTestSpace) ... ok
testCollisionHandlerPostSolve (test_space.UnitTestSpace) ... ok
testCollisionHandlerPreSolve (test_space.UnitTestSpace) ... ok
testCollisionHandlerPreSolveNoReturn (test_space.UnitTestSpace) ... ok
testCollisionHandlerSeparate (test_space.UnitTestSpace) ... ok
testCopyMethods (test_space.UnitTestSpace) ... ok
testDebugDraw (test_space.UnitTestSpace) ... ok
testDefaultCollisionHandler (test_space.UnitTestSpace) ... ok
testPointQuery (test_space.UnitTestSpace) ... ok
testPointQueryNearest (test_space.UnitTestSpace) ... ok
testPointQueryNearestSensor (test_space.UnitTestSpace) ... ok
testPointQueryNearestWithShapeFilter (test_space.UnitTestSpace) ... ok
testPointQuerySensor (test_space.UnitTestSpace) ... ok
testPostStepCallback (test_space.UnitTestSpace) ... ok
testProperties (test_space.UnitTestSpace) ... ok
testReindexShape (test_space.UnitTestSpace) ... ok
testReindexShapesForBody (test_space.UnitTestSpace) ... ok
testReindexStatic (test_space.UnitTestSpace) ... ok
testReindexStaticCollision (test_space.UnitTestSpace) ... ok
testRemoveInStep (test_space.UnitTestSpace) ... ok
testSegmentQuery (test_space.UnitTestSpace) ... ok
testSegmentQueryFirst (test_space.UnitTestSpace) ... ok
testSegmentQueryFirstSensor (test_space.UnitTestSpace) ... ok
testSegmentQuerySensor (test_space.UnitTestSpace) ... ok
testShapeQuery (test_space.UnitTestSpace) ... ok
testShapeQuerySensor (test_space.UnitTestSpace) ... ok
testStaticPointQueries (test_space.UnitTestSpace) ... ok
testStaticSegmentQueries (test_space.UnitTestSpace) ... ok
testThreaded (test_space.UnitTestSpace) ... ok
testWildcardCollisionHandler (test_space.UnitTestSpace) ... ok
testCoordinateConversion (test_body.UnitTestBody) ... ok
testDynamic (test_body.UnitTestBody) ... ok
testEachArbiters (test_body.UnitTestBody) ... ok
testForce (test_body.UnitTestBody) ... ok
testGetConstraints (test_body.UnitTestBody) ... ok
testGetShapes (test_body.UnitTestBody) ... ok
testImpulse (test_body.UnitTestBody) ... ok
testKinematic (test_body.UnitTestBody) ... ok
testKineticEnergy (test_body.UnitTestBody) ... ok
testMassMomentFromShape (test_body.UnitTestBody) ... ok
testPickle (test_body.UnitTestBody) ... 1
2
3
4
5
6
ok
testPositionFunction (test_body.UnitTestBody) ... ok
testProperties (test_body.UnitTestBody) ... ok
testRepr (test_body.UnitTestBody) ... ok
testSleep (test_body.UnitTestBody) ... ok
testSleepWithGroup (test_body.UnitTestBody) ... ok
testStatic (test_body.UnitTestBody) ... ok
testVelocityConversion (test_body.UnitTestBody) ... ok
testVelocityFunction (test_body.UnitTestBody) ... ok
testContactPointSet (test_arbiter.UnitTestArbiter) ... ok
testFriction (test_arbiter.UnitTestArbiter) ... ok
testImpulse (test_arbiter.UnitTestArbiter) ... ok
testIsFirstContact (test_arbiter.UnitTestArbiter) ... ok
testIsRemoval (test_arbiter.UnitTestArbiter) ... ok
testRestitution (test_arbiter.UnitTestArbiter) ... ok
testShapes (test_arbiter.UnitTestArbiter) ... ok
testSurfaceVelocity (test_arbiter.UnitTestArbiter) ... ok
testTotalKE (test_arbiter.UnitTestArbiter) ... ok
testCircleBB (test_shape.UnitTestCircle) ... ok
testCircleNoBody (test_shape.UnitTestCircle) ... ok
testOffset (test_shape.UnitTestCircle) ... ok
testOffsetUnsafe (test_shape.UnitTestCircle) ... ok
testPickle (test_shape.UnitTestCircle) ... ok
testRadius (test_shape.UnitTestCircle) ... ok
testRadiusUnsafe (test_shape.UnitTestCircle) ... ok
testBB (test_shape.UnitTestPoly) ... ok
testCreateBox (test_shape.UnitTestPoly) ... ok
testInit (test_shape.UnitTestPoly) ... ok
testPickle (test_shape.UnitTestPoly) ... ok
testRadius (test_shape.UnitTestPoly) ... ok
testRadiusUnsafe (test_shape.UnitTestPoly) ... ok
testVertices (test_shape.UnitTestPoly) ... ok
testVerticesUnsafe (test_shape.UnitTestPoly) ... ok
testBB (test_shape.UnitTestSegment) ... ok
testPickle (test_shape.UnitTestSegment) ... ok
testProperties (test_shape.UnitTestSegment) ... ok
testPropertiesUnsafe (test_shape.UnitTestSegment) ... ok
testSegmentSegmentCollision (test_shape.UnitTestSegment) ... ok
testSetNeighbors (test_shape.UnitTestSegment) ... ok
testArea (test_shape.UnitTestShape) ... ok
testCenterOfGravity (test_shape.UnitTestShape) ... ok
testCollisionType (test_shape.UnitTestShape) ... ok
testDensity (test_shape.UnitTestShape) ... ok
testElasticity (test_shape.UnitTestShape) ... ok
testFilter (test_shape.UnitTestShape) ... ok
testFriction (test_shape.UnitTestShape) ... ok
testMass (test_shape.UnitTestShape) ... ok
testMoment (test_shape.UnitTestShape) ... ok
testNoBody (test_shape.UnitTestShape) ... ok
testPickle (test_shape.UnitTestShape) ... ok
testPointQuery (test_shape.UnitTestShape) ... ok
testRemoveBody (test_shape.UnitTestShape) ... ok
testSegmentQuery (test_shape.UnitTestShape) ... ok
testSensor (test_shape.UnitTestShape) ... ok
testShapesCollide (test_shape.UnitTestShape) ... ok
testSpace (test_shape.UnitTestShape) ... ok
testSurfaceVelocity (test_shape.UnitTestShape) ... ok
testSwitchBody (test_shape.UnitTestShape) ... ok
testCreation (test_bb.UnitTestBB) ... ok
testMerge (test_bb.UnitTestBB) ... ok
testMethods (test_bb.UnitTestBB) ... ok
testPickle (test_bb.UnitTestBB) ... ok
test_convex_decomposition (test_autogeometry.UnitTestAutoGeometry) ... ok
test_is_closed (test_autogeometry.UnitTestAutoGeometry) ... ok
test_march_hard (test_autogeometry.UnitTestAutoGeometry) ... ok
test_march_soft (test_autogeometry.UnitTestAutoGeometry) ... ok
test_simplify_curves (test_autogeometry.UnitTestAutoGeometry) ... ok
test_simplify_vertexes (test_autogeometry.UnitTestAutoGeometry) ... ok
test_to_convex_hull (test_autogeometry.UnitTestAutoGeometry) ... ok
test_collect_segment (test_autogeometry.UnitTestPolylineSet) ... ok
testA (test_constraint.UnitTestConstraint) ... ok
testActivate (test_constraint.UnitTestConstraint) ... ok
testB (test_constraint.UnitTestConstraint) ... ok
testCollideBodies (test_constraint.UnitTestConstraint) ... ok
testErrorBias (test_constraint.UnitTestConstraint) ... ok
testImpulse (test_constraint.UnitTestConstraint) ... ok
testMaxBias (test_constraint.UnitTestConstraint) ... ok
testMaxForce (test_constraint.UnitTestConstraint) ... ok
testPickle (test_constraint.UnitTestConstraint) ... ok
testDamping (test_constraint.UnitTestDampedRotarySpring) ... ok
testPickle (test_constraint.UnitTestDampedRotarySpring) ... ok
testRestAngle (test_constraint.UnitTestDampedRotarySpring) ... ok
testStiffness (test_constraint.UnitTestDampedRotarySpring) ... ok
testAnchor (test_constraint.UnitTestDampedSpring) ... ok
testDamping (test_constraint.UnitTestDampedSpring) ... ok
testPickle (test_constraint.UnitTestDampedSpring) ... ok
testRestLength (test_constraint.UnitTestDampedSpring) ... ok
testStiffness (test_constraint.UnitTestDampedSpring) ... ok
testPhase (test_constraint.UnitTestGearJoint) ... ok
testPickle (test_constraint.UnitTestGearJoint) ... ok
testRatio (test_constraint.UnitTestGearJoint) ... ok
testAnchor (test_constraint.UnitTestGrooveJoint) ... ok
testGroove (test_constraint.UnitTestGrooveJoint) ... ok
testPickle (test_constraint.UnitTestGrooveJoint) ... ok
testAnchor (test_constraint.UnitTestPinJoint) ... ok
testDistane (test_constraint.UnitTestPinJoint) ... ok
testPickle (test_constraint.UnitTestPinJoint) ... ok
testAnchorByAnchor (test_constraint.UnitTestPivotJoint) ... ok
testAnchorByPivot (test_constraint.UnitTestPivotJoint) ... ok
testPickle (test_constraint.UnitTestPivotJoint) ... ok
testAngle (test_constraint.UnitTestRatchetJoint) ... ok
testPhase (test_constraint.UnitTestRatchetJoint) ... ok
testPickle (test_constraint.UnitTestRatchetJoint) ... ok
testRatchet (test_constraint.UnitTestRatchetJoint) ... ok
testMax (test_constraint.UnitTestRotaryLimitJoint) ... ok
testMin (test_constraint.UnitTestRotaryLimitJoint) ... ok
testPickle (test_constraint.UnitTestRotaryLimitJoint) ... ok
testPickle (test_constraint.UnitTestSimleMotor) ... ok
testSimpleMotor (test_constraint.UnitTestSimleMotor) ... ok
testAnchor (test_constraint.UnitTestSlideJoint) ... ok
testMax (test_constraint.UnitTestSlideJoint) ... ok
testMin (test_constraint.UnitTestSlideJoint) ... ok
testPickle (test_constraint.UnitTestSlideJoint) ... ok
testConstants (test_shape_filter.UnitTestShapeFilter) ... ok
testEq (test_shape_filter.UnitTestShapeFilter) ... ok
testInit (test_shape_filter.UnitTestShapeFilter) ... ok
testPickle (test_shape_filter.UnitTestShapeFilter) ... ok
testManyBoxCrash (test_common.UnitTestBugs) ... ok
testNoStaticShape (test_common.UnitTestBugs) ... ok
testAreaHelpers (test_common.UnitTestGeneral) ... ok
testGeneral (test_common.UnitTestGeneral) ... ok
testMomentHelpers (test_common.UnitTestGeneral) ... ok
testAnglesDegrees (test_vec2d.UnitTestVec2d) ... ok
testAnglesRadians (test_vec2d.UnitTestVec2d) ... ok
testComparison (test_vec2d.UnitTestVec2d) ... ok
testCreationAndAccess (test_vec2d.UnitTestVec2d) ... ok
testCross (test_vec2d.UnitTestVec2d) ... ok
testHighLevel (test_vec2d.UnitTestVec2d) ... ok
testInplace (test_vec2d.UnitTestVec2d) ... ok
testLength (test_vec2d.UnitTestVec2d) ... ok
testMath (test_vec2d.UnitTestVec2d) ... ok
testPickle (test_vec2d.UnitTestVec2d) ... ok
testReverseMath (test_vec2d.UnitTestVec2d) ... ok
testUnary (test_vec2d.UnitTestVec2d) ... ok

----------------------------------------------------------------------
Ran 182 tests in 0.166s

OK
```

**Comment by viblo** [comment]
Thanks for the report.. Seems like something broke with the latest release of cffi. But when I tried on cpython 3.7 it worked fine with cffi 1.13.1 (except for deprecation warnings)

**Comment by viblo** [comment]
Seems to be linux problem, I cant reproduce on windows, even with cpython 2.7

**Comment by manuelli** [comment]
It's possible that it's Linux specific, all my tests were done on Linux. Should be easy to test by creating a docker image based on 16.04 and running the test in there. Do any of the travis builds (https://travis-ci.org/viblo/pymunk) target Linux, or run in a linux docker container?

**Comment by viblo** [comment]
Yes, the linux wheels are built in a linux container. I will test it later today.

**Comment by viblo** [comment]
Can confirm that I get the same error as you on linux. Seems to be something in cffi 1.13.1 triggering it, because 1.13 works fine for me.

**Comment by viblo** [comment]
I opened a issue in the cffi issue tracker since I couldnt see any obvious things about what can be done on pymunk side: https://bitbucket.org/cffi/cffi/issues/432/1131-breaks-pymunk-on-linux-in-a-subtle

**Comment by manuelli** [comment]
Ok thanks. I think a temporary fix on pymunk side could be to require `cffi <= 1.13` since then at least when someone does `pip install pymunk` on Linux it won't be broken straight out of the box.

**Comment by viblo** [comment]
Fixed both by latest pymunk which avoids 1.13.1, and by a new release of cffi.
